src/runtime/GlobalObjectBuiltinIntl.cpp

```diff
@@ -2539,6 +2539,18 @@ static Value builtinIntlNumberFormatSupportedLocalesOf(ExecutionState& state, Va
     return supportedLocales(state, availableLocales, requestedLocales, options);
 }
 
+static Value builtinIntlGetCanonicalLocales(ExecutionState& state, Value thisValue, size_t argc, Value* argv, bool isNewExpression)
+{
+    // Let ll be ? CanonicalizeLocaleList(locales).
+    ValueVector ll = canonicalizeLocaleList(state, argv[0]);
+    // Return CreateArrayFromList(ll);
+    ArrayObject* ret = new ArrayObject(state, 0.0);
+    for (size_t i = 0; i < ll.size(); i++) {
+        ret->defineOwnProperty(state, ObjectPropertyName(state, Value(i)), ObjectPropertyDescriptor(Value(ll[i]), ObjectPropertyDescriptor::AllPresent));
```

Comment:
For `Value(ll[i])`, do we have to generate a temporal Value? How about directly passing the element like `ObjectPropertyDescriptor(ll[i], ...)`?

Reply:
Indeed we can pass it like `ObjectPropertyDescriptor(ll[i], ...)`, thanks.